aiohttp/http_writer.py

```diff
@@ -71,10 +70,10 @@ def _write(self, chunk: bytes) -> None:
         size = len(chunk)
         self.buffer_size += size
         self.output_size += size
-
-        if self._transport is None or self._transport.is_closing():
+        transport = self.transport
+        if not self._protocol.connected or transport is None or transport.is_closing():
```

Comment:
Surely the other 2 checks would always produce the same result as the first? Do we actually need them anymore?

Reply:
Still not clear if these are needed, but will leave them to be safe.